sdk/nanotdf.go

```diff
@@ -669,18 +673,28 @@ func (s SDK) CreateNanoTDF(writer io.Writer, reader io.Reader, config NanoTDFCon
 		return 0, errors.New("exceeds max size for nano tdf")
 	}
 
-	kasURL, err := config.kasURL.getURL()
+	kasURL, err := config.kasURL.GetURL()
 	if err != nil {
 		return 0, fmt.Errorf("config.kasURL failed:%w", err)
 	}
 
-	kasPublicKey, err := getECPublicKey(kasURL, s.dialOptions...)
+	kasPublicKey, kid, err := getECPublicKeyKid(kasURL, s.dialOptions...)
 	if err != nil {
 		return 0, fmt.Errorf("getECPublicKey failed:%w", err)
 	}
-
 	slog.Debug("CreateNanoTDF", slog.String("header size", kasPublicKey))
 
+	// kid from kasPublicKey endpoint
+	slog.Debug("kasPublicKey", slog.String("kid", kid))
+
+	// update KAS URL with kid if set
+	if kid != "" {
```

Comment:
You'll probably need to guard this somehow to work with existing servers with a command line flag, or just defer this to a later PR

```suggestion
	if kid != "" && newKIDEmbeddingEnabled {
```

Reply:
since we aren't bumping the NanoTDF version this will need to work with pre- and post-feature servers

Reply:
now checking `c.tdfFeatures.noNanoKID`